src/Voron/Data/Compression/DecompressionBuffersPool.cs

```diff
@@ -115,6 +115,7 @@ public IDisposable GetTemporaryPage(LowLevelTransaction tx, int pageSize, out Te
                 {
                     buffer.EnsureValidPointer(tx);
                     tmp = buffer.TempPage;
+                    break;
```

Comment:
Wouldn't using an if statement be more readable than a while statement if this is indeed the right logic?
if(queue.TryDequeue(out buffer) && buffer.CanReuse)
{
...
}